nixos/modules/security/pam.nix

```diff
@@ -1639,6 +1640,13 @@ in
 
     security.pam.package = lib.mkPackageOption pkgs "pam" { };
 
+    security.pam.pam_unixModulePath = lib.mkOption {
+      type = lib.types.pathInStore;
+      default = "${package}/lib/security/pam_unix.so";
+      defaultText = "\${config.security.pam.package}/lib/security/pam_unix.so";
+      description = "The pam_unix module to use in all the default pam services.";
+    };
```

Comment:
What do you think about making this option `internal`? That way, we only have to work around in-tree usages as we refactor the rule stack.

Reply:
That is a valid point. I briefly poked the whole pam mess, but it needs a lot of rework. Cleaning this up is definitely something we should do, and making that as little pain as possible is a good idea. Will do.